radio/src/targets/horus/tp_gt911.cpp

```diff
@@ -634,25 +644,36 @@ bool touchPanelInit(void)
     {
       TRACE("GT911 chip detected");
       tmp[0] = 0X02;
-      if (!I2C_GT911_WriteRegister(GT_CTRL_REG, tmp, 1))
+      if (!I2C_GT911_WriteRegister(GT911_COMMAND_REG, tmp, 1))
       {
-        TRACE("GT911 ERROR: write to control register failed");
+        TRACE("GT911 ERROR: write to command register failed");
       }
-      if (!I2C_GT911_ReadRegister(GT_CFGS_REG, tmp, 1))
+      if (!I2C_GT911_ReadRegister(GT911_CONFIG_REG, tmp, 1))
       {
           TRACE("GT911 ERROR: configuration register read failed");
       }
 
       TRACE("Chip config Ver:%x", tmp[0]);
-      if (tmp[0] < GT911_CFG_NUMER)  //Config ver
+      if (tmp[0] < GT911_CFG_NUMBER)  //Config ver
       {
-        TRACE("Sending new config %d", GT911_CFG_NUMER);
+        TRACE("Sending new config %d", GT911_CFG_NUMBER);
         if (!I2C_GT911_SendConfig())
         {
           TRACE("GT911 ERROR: sending configration failed");
         }
       }
 
+      // workaround: cannot reset config number
+      if (!I2C_GT911_ReadRegister(GT911_COORDINATE_REG, tmp, 1))
+      {
+        TRACE("GT911 ERROR: reading coordinate register failed");
+      }
+      else {
+          if (tmp[0] != GT911_COOR_CONFIG_VALUE) {
+              reverseCoordinates = true;
```

Comment:
This is assuming that the only change in GT911_COORDINATE_REG was the coordinate inversion (0x3C vs 0xFC).
I would suggest instead, for clarity and also for being more correct, just to reverse the coordinates for X12s, without defining custom GT911_COOR_CONFIG_VALUE for X12s in the first place, if writing the GT911 config is not working.
E.g.:
```
#if defined(PCBX12S)
  reverseCoordinates = true;
#endif
```
instead of the above code trying to write GT911_COORDINATE_REG, but failing and then resorting to the trick here.

Reply:
Thanks for the suggestion!
Now I removed the runtime check completely and simple use compiletime decision for X12S.